core/src/main/java/org/elasticsearch/rest/RestHandler.java

```diff
@@ -46,4 +47,12 @@ default boolean canTripCircuitBreaker() {
     default boolean supportsPlainText() {
         return false;
     }
+
+    /**
+     * Indicates if the RestHandler supports content as a stream. A stream would be multiple objects delineated by
+     * {@link XContent#streamSeparator()}
```

Comment:
Maybe add that based on what this method returns we decide also which content types to accept?